src/alchemlyb/visualisation/ti_dhdl.py

```diff
@@ -60,6 +59,20 @@ def plot_ti_dhdl(dhdl_data, labels=None, colors=None, units='kT',
         for dhdl in dhdl_data:
             dhdl_list.extend(dhdl.separate_dhdl())
 
+    # Convert unit
+    new_unit = []
+    for dhdl in dhdl_list:
+        _converters = {'kT': to_kT, 'kJ/mol': to_kJmol,
+                       'kcal/mol': to_kcalmol}
```

Comment:
define outside of the loop ... or just define a function  it in `postprocessors.units`
```python
def get_unit_converter(unit):
   try:
       return _converters[unit]
   except KeyError:
       raise ValueError(...)
```